dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1075,10 +1185,17 @@ CDefaultContextMenu::TryToBrowse(
 
     /* Get a pointer to the shell browser */
     hr = IUnknown_QueryService(m_site, SID_IShellBrowser, IID_PPV_ARG(IShellBrowser, &psb));
-    if (FAILED_UNEXPECTEDLY(hr))
-        return 0;
+    if (FAILED(hr))
+        return hr;
 
-    return psb->BrowseObject(ILCombine(m_pidlFolder, pidl), wFlags);
+    PIDLIST_ABSOLUTE pidl;
```

Comment:
If you'd use `CComHeapPtr<ITEMIDLIST>` here, then it will be cleaned up automatically.

Reply:
That is true but there are only 4 lines between the alloc and free here, it should be pretty clear what is going on.